### PR TITLE
Support for updating signed Logitech unifying receivers

### PR DESCRIPTION
**Pull request by SonicFrog**
Your mail reminded me this has been hanging on my hard drive since December.

This implementes updating signed logitech unifying receivers, both for RQR12 and RQR24

Signed-off-by: Ogier Bouvier <obouvier@logitech.com>

### CODE REVIEW

File: plugins/unifying/lu-device-bootloader-nordic.c [hunk 1]
```diff
@@ -209,12 +228,24 @@ lu_device_bootloader_nordic_write_firmware (LuDevice *device, GBytes *fw, GError
...
-							payload->data,
-							error))
+
+		if (payload->cmd == LU_DEVICE_BOOTLOADER_CMD_WRITE_SIGNATURE)
+			res = lu_device_bootloader_nordic_write_signature(device,
+									  payload->addr,
+									  payload->len,
+									  payload->data,
+									  error);
+		else {
```
Comment (hughsie) [inline plugins/unifying/lu-device-bootloader-nordic.c]: brace both bits please, e.g. `} else {`

File: plugins/unifying/lu-device-bootloader-nordic.c [hunk 2]
```diff
@@ -228,14 +259,14 @@ lu_device_bootloader_nordic_write_firmware (LuDevice *device, GBytes *fw, GError
...
-	/* set the reset vector */
 	if (!lu_device_bootloader_nordic_write (device,
-						0x0000,
+						0x000,
 						0x01,
 						payload->data,
 						error))
 		return FALSE;
 
+
```
Comment (hughsie) [inline plugins/unifying/lu-device-bootloader-nordic.c]: double newline.

File: plugins/unifying/lu-device-bootloader-nordic.c [hunk 3]
```diff
@@ -228,14 +259,14 @@ lu_device_bootloader_nordic_write_firmware (LuDevice *device, GBytes *fw, GError
 						error))
 		return FALSE;
 
-	/* set the reset vector */
 	if (!lu_device_bootloader_nordic_write (device,
-						0x0000,
+						0x000,
```
Comment (hughsie) [inline plugins/unifying/lu-device-bootloader-nordic.c]: Isn't this 16 bits wide?

File: plugins/unifying/lu-device.h [hunk 4]
```diff
@@ -64,7 +64,8 @@ struct _LuDeviceClass
 
 #define LU_DEVICE_EP1				0x81
 #define LU_DEVICE_EP3				0x83
-#define LU_DEVICE_TIMEOUT_MS			2500
+/* Signed firmware are very long to verify on the device */
+#define LU_DEVICE_TIMEOUT_MS			20000
```
Comment (hughsie) [inline plugins/unifying/lu-device.h]: This is fine, but do we set the device status to `FWUPD_STATUS_DEVICE_BUSY` when this is happening?

File: plugins/unifying/lu-device-bootloader-nordic.c (hunk 3 above)
Comment (SonicFrog) [reply plugins/unifying/lu-device-bootloader-nordic.c]: Indeed, my bad

File: plugins/unifying/lu-device-bootloader.c [hunk 5]
```diff
@@ -88,9 +98,16 @@ lu_device_bootloader_parse_requests (LuDevice *device, GBytes *fw, GError **erro
 					     payload->len);
 				return NULL;
 			}
+
```
Comment (superm1) [inline plugins/unifying/lu-device-bootloader.c]: unneeded newline here

File: plugins/unifying/lu-device.h (hunk 4 above)
Comment (SonicFrog) [reply plugins/unifying/lu-device.h]: Wasn't this already done? Even when the timeout was only 2.5 seconds?

**Pull request merged**